### PR TITLE
Performance updates in self.update_user_groups method

### PR DESCRIPTION
**Pull request by theognis1002**
Made some very slight performance updates in the self.update_user_groups method. `values_list()` retrieves a subset of data without the overhead of creating model instances

### CODE REVIEW

**Comment by sondrelg** [comment]
The `values_list` changes look good, but can you explain why it makes sense to replace `set` with `add`? I'm not that familiar with this code, so just want to know your reasoning before diving in further 🙂

**Comment by JonasKs** [comment]
Group tests fails, so I'm not sure if this works as intended?

**Comment by theognis1002** [comment]
@JonasKs ahh thx! didn't see that they failed at the time. I reverted it back to set.

**Comment by JonasKs** [comment]
This looks good to me. 

@sondrelg, @tim-schilling , opinions? 😊

**Comment by sondrelg** [comment]
Could you please explain how
```python
list(existing_groups.iterator()) + new_groups
```
is more performant than this?
```python
existing_groups + new_groups
```

I'm sure it could be, but to me it just looks like extra steps 🙂

**Comment by theognis1002** [comment]
@sondrelg `list(existing_groups.iterator())` was just taken from L213 and moved down. This was so that the queryset would not be evaluated before it was needed

**Comment by sondrelg** [comment]
Ah yeah I see 👍 

That being said, I'm not sure the `iterator()` call is worth keeping. Doing a little testing, it seems like the `list` call is all that's needed.

```python
from django.contrib.auth.models import Group

g = Group.objects.all()

print(list(g) == list(g.iterator()))

>> True
```

I could be wrong though 🙂 

Another alternative to combining lists here might be to merge querysets (if `new_groups`'s queryset is available), using the pipe (`|`) operator.

Otherwise, looks good to me!

**Comment by tim-schilling** [comment]
This looks good to me. It reminds me that I need to find time to do that benchmarking on #220

**Comment by JonasKs** [comment]
Hey @theognis1002 , could you address @sondrelg's comment so we can merge? 😊

File: django_auth_adfs/backend.py [hunk 1]
```diff
@@ -222,8 +222,8 @@ def update_user_groups(self, user, claims):
                              settings.GROUPS_CLAIM)
                 claim_groups = []
             if sorted(claim_groups) != sorted(django_groups):
-                existing_groups = list(Group.objects.filter(name__in=claim_groups).iterator())
-                existing_group_names = frozenset(group.name for group in existing_groups)
+                existing_groups = Group.objects.filter(name__in=claim_groups)
+                existing_group_names = frozenset(existing_groups.values_list("name", flat=True))
```
Comment (stephane) [inline django_auth_adfs/backend.py:226]: This `values_list` call will generate a SQL query and the lazy`existing_groups` is casted to list later (one query), the new code adds a new SQL query.

**Pull request closed without merging**